var/spack/repos/builtin/packages/gptune/package.py

```diff
@@ -214,28 +184,89 @@ def test(self):
                 + '{\\"nodes\\":$nodes,\\"cores\\":$cores}}}") \n'
             )
 
-        # copy the environment configuration files to non-cache directories
-        op = ["run_env.sh", python_platlib + "/gptune/."]
-        self.run_test("cp", options=op, work_dir=wd)
-        op = ["run_env.sh", self.install_test_root + "/."]
-        self.run_test("cp", options=op, work_dir=wd)
-
-        apps = ["Scalapack-PDGEQRF_RCI"]
-        if "+mpispawn" in spec:
-            apps = apps + ["GPTune-Demo", "Scalapack-PDGEQRF"]
-        if "+superlu" in spec:
-            apps = apps + ["SuperLU_DIST_RCI"]
-            if "+mpispawn" in spec:
-                apps = apps + ["SuperLU_DIST"]
-        if "+hypre" in spec:
-            if "+mpispawn" in spec:
-                apps = apps + ["Hypre"]
+        # copy the environment configuration to the python install directory
+        cp = which("cp")
+        cp(script_path, join_path(python_platlib, "gptune"))
+
+    def setup_run_environment(self, env):
+        env.set("GPTUNE_INSTALL_PATH", python_platlib)
+
+    bash = which("bash")
+    cp = which("cp")
+    git = which("git")
+    rm = which("rm")
+
+    def test_hypre(self):
+        """set up and run hypre example"""
+        if "~hypre" in self.spec or "~mpispawn" in self.spec:
+            raise SkipTest("Package must be installed with +hypre+mpispawn")
+
+        test_dir = join_path(self.test_suite.current_test_cache_dir, self.examples_src_dir)
+
+        # copy hypre executables to the correct place
+        # TBD: Is it safe to not have the version match that of the build?
+        wd = join_path(test_dir, "Hypre")
+        with working_dir(wd):
+            self.rm("-rf", "hypre")
+            self.git("clone", "https://github.com/hypre-space/hypre.git")
```

Comment:
Could we put the version into the clone commands for these? There is a `f"v{self.spec['hypre'].version}"` tag for each numeric version of `hypre`. Looks like the same is true for `superlu-dist`.

Reply:
The hypre version we use for gptune testing is v2.19.0, newer versions will break the test. For superlu_dist, all recent versions (after v7.0.0) work with the gptune testing.

Reply:
@liuyangzhuan @becker33 

I modified cloning within the stand-alone tests to clone only the desired version of the package, adding `SkipTest` exceptions to respect the version constraints.  

I am also using a "hammer" in the form of modifying all of the associated bash scripts for a test to `set -e` (i.e., terminate on first command failure) so I could remove the `timeout` option.  IMO This change means we can re-enable stand-alone testing of `gptune`, though it does appear the tests will be failing until the underlying `numpy`- and `jq`-related changes are made to the files used for testing.